db/hints/internal/hint_sender.cc

```diff
@@ -50,8 +50,8 @@ namespace internal {
 
 namespace {
 
-[[maybe_unused]] std::optional<gms::inet_address> maybe_convert_hid_to_ep_with_erm(
-        locator::effective_replication_map_ptr ermptr, const locator::host_id& hid) {
+std::optional<gms::inet_address> maybe_convert_hid_to_ep_with_erm(locator::effective_replication_map_ptr ermptr,
```

Comment:
Why we need this helper? It's used only in one place and doesn't bring much value. And in `hint_sender::can_send` we use `get_endpoint_for_host_id_if_known` directly.